ui/src/message_popup/author/edit_question_page.jsx

```diff
@@ -11,15 +11,96 @@ export default React.createClass({
   displayName: 'EditQuestionPage',
 
   propTypes: {
-    questionId: React.PropTypes.string.isRequired
+    questionId: React.PropTypes.string.isRequired,
+    loaded: React.PropTypes.bool,
+    allQuestions: React.PropTypes.object
+  },
+
+  getDefaultProps(){
+    return ({
+      loaded: false,
+      allQuestions: {
+        currentQuestions: [],
+        archivedQuestions: []
+      }
+    });
+  },
+
+  getInitialState(){
+    return ({
+      loaded: this.props.loaded,
+      allQuestions: this.props.allQuestions
+    });
+  },
+
+  componentDidMount(){
+    this.queryDatabase();
+  },
+
+  queryDatabase(){
+    Api.questionsQuery().end(this.onQuestionsReceived);
+  },
+
+  getNewID(){
+    const {allQuestions} = this.state;
+    const allQuestionsFlat = allQuestions.currentQuestions.concat(allQuestions.archivedQuestions);
+    const mostRecentQuestion = _.maxBy(allQuestionsFlat, question => question.id);
+    const largestID = mostRecentQuestion !== undefined ? mostRecentQuestion.id : 0;
+    return largestID + 1;
+  },
+
+  onQuestionsReceived(err, response){
+    if(err){
+      this.setState({loaded: true});
+      return;
+    }
+    const allQuestions = JSON.parse(response.text).questions;
+    this.setState({ loaded: true, allQuestions });
+  },
+
+  onEditQuestion(newQuestion, originalQuestion){
+    const {currentQuestions, archivedQuestions} = this.state.allQuestions;
+    const question = {id: this.getNewID(), ...newQuestion};
+    const newCurrentQuestions = currentQuestions.filter(question => question.id !== originalQuestion.id).concat(question);
+    const newArchivedQuestions = archivedQuestions.concat(originalQuestion);
+    console.log('Replacing...');
```

Comment:
I thought about doing so but it will be getting redirected to `QuestionsPage` immediately after so I felt that updating the state was unnecessary. `QuestionsPage` should query the database and get the updated questions list.